gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp

```diff
@@ -750,36 +758,61 @@ OGRMultiLineString *OGRFlatGeobufLayer::readMultiLineString(const Feature *featu
 OGRLinearRing *OGRFlatGeobufLayer::readLinearRing(const Feature *feature, uint32_t len, uint32_t offset)
 {
     auto lr = new OGRLinearRing();
-    readSimpleCurve(feature, len, offset, lr);
+    if (readSimpleCurve(feature, len, offset, lr) != OGRERR_NONE) {
+        delete lr;
+        return nullptr;
+    }
     return lr;
 }
 
-void OGRFlatGeobufLayer::readSimpleCurve(const Feature *feature, uint32_t len, uint32_t offset, OGRSimpleCurve *sc)
+OGRErr OGRFlatGeobufLayer::readSimpleCurve(const Feature *feature, uint32_t len, uint32_t offset, OGRSimpleCurve *sc)
 {
-    auto xy = feature->xy()->data();
+    if (len > std::numeric_limits<int>().max())
```

Comment:
`std::numeric_limits<int>::max()` is a bit more usual I think. You'll probably get a warning about signed / unsigned comparison, so you need to add an explicit cast: `static_cast<uint32_t>(std::numeric_limits<int>::max())`